be/src/exec/pipeline/pipeline_driver_queue.cpp

```diff
@@ -2,80 +2,158 @@
 
 #include "exec/pipeline/pipeline_driver_queue.h"
 
+#include "exec/workgroup/work_group.h"
 #include "gutil/strings/substitute.h"
+
 namespace starrocks::pipeline {
-void QuerySharedDriverQueue::close() {
+
+void QuerySharedDriverQueue::put_back(const DriverRawPtr driver, bool from_dispatcher) {
```

Comment:
two functions have different semantics may have different names. e.g.
put_back_from_dispatcher
put_back
so the code would be understandable and clear

Reply:
Sounds great.